### PR TITLE
ISSUE: https://github.com/SAP/yeoman-ui/issues/288

### PR DESCRIPTION
**Pull request by tomer-epstein**

### CODE REVIEW

File: src/Form.vue [hunk 1]
```diff
@@ -204,13 +204,20 @@ export default {
       switch (question.type) {
         case "list":
         case "rawlist":
+          if (question._default === undefined) {
+            question.answer = undefined;
```
Comment (perspectivus1) [inline src/Form.vue:208]: You're inside a method called `getInitialAnswer()`. Don't set the `question.answer` here -- return the answer.

I believe this entire case (`list` and `rawlist`) can be written like this:
```js
if (question._choices.length === 0 || question._default === undefined) {
    this.setInvalid(question, NOT_ANSWERED);
    return;
}
if (typeof question._default === "number" && question._default < question._choices.length) {
    index = question._default;
}
if (index < 0) {
    index = question._choices.findIndex(choice => {
        if (question._default) {
            return choice.value === question._default;
        }
    });
}
if (index < 0) {
    this.setInvalid(question, NOT_ANSWERED);
    return;
} else {
    return question._choices[index].value;
}
```

... but please verify

File: src/Form.vue [hunk 2]
```diff
@@ -346,6 +353,20 @@ export default {
               }
             }
 
+            // evaluate default()
```
Comment (perspectivus1) [inline src/Form.vue:356]: Reversing the order of evaluating `default()` and `choices()` could cause issues:
For example, you can tell that a default value of 10 is invalid if there are only 9 items in the list; therefore, you must evaluate the choices before the evaluating the default value.

File: src/Form.vue (hunk 1 above)
Comment (tomer-epstein) [reply src/Form.vue]: done

File: src/Form.vue (hunk 2 above)
Comment (tomer-epstein) [reply src/Form.vue:353]: I couldn't find in inquirer.js documentation  that they mention the functions evaluate order.

image

Comment (tomer-epstein) [reply src/Form.vue:353]: done

File: jest.config.js [hunk 3]
```diff
@@ -51,8 +51,8 @@ module.exports = {
     "global": {
       "branches": 85,
       "functions": 90,
-      "lines": 90,
-      "statements": 90
+      "lines": 89,
+      "statements": 89
```
[the remark this answers is not in the document]
Comment (tomer-epstein) [reply jest.config.js]: done

**Pull request merged**